### PR TITLE
Check space left before downloading files

### PR DESCRIPTION
**Pull request by tevincent**
https://github.com/user-attachments/assets/756d63bd-c11d-4d24-977e-20d82a034a29

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/data/services/BulkDownloadWorker.kt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/data/services/BulkDownloadWorker.kt:1]: There is a `WIP` commit in this branch, can you remove or rename it ?

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/data/services/BulkDownloadWorker.kt [hunk 2]
```diff
@@ -34,9 +35,13 @@ import com.infomaniak.drive.utils.NotificationUtils.cancelNotification
...
 
     private val folderId: Int by lazy { inputData.getInt(FOLDER_ID, 0) }
-    private val fileIds: List<Int> by lazy {
-        FileController.getFolderOfflineFilesId(folderId = folderId, UiSettings(context).sortType)
+    private val files: List<File> by lazy {
+        FileController.getFolderOfflineFilesId(folderId = folderId, UiSettings(context).sortType).map {
+            val file = FileController.getFileById(it, userDrive)!!
+            file
+        }
     }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/data/services/BulkDownloadWorker.kt:43]:
```suggestion

    private val files: List<File> by lazy {
        FileController.getFolderOfflineFilesId(folderId = folderId, sortType = UiSettings(context).sortType)
            .map { FileController.getFileById(it, userDrive)!! }
    }
```

File: app/src/main/java/com/infomaniak/drive/utils/DownloadOfflineFileManager.kt [hunk 3]
```diff
@@ -226,10 +227,10 @@ class DownloadOfflineFileManager(
     companion object {
         private const val MAX_INTERVAL_BETWEEN_PROGRESS_UPDATE_MS = 1000L
 
-        fun observeBulkDownloadOffline(context: Context) = WorkManager.getInstance(context).getWorkInfosLiveData(
+        fun observeFailureDownloadWorkerOffline(context: Context) = WorkManager.getInstance(context).getWorkInfosLiveData(
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/utils/DownloadOfflineFileManager.kt:230]: This function is named `observe…()`, but it doesn't observe anything.
It's just creating the liveData that is then observed in the calling method.

Can you rename it ?
Maybe something in the likes of `getFailureDownloadWorkerOffline()` ?

File: app/src/main/java/com/infomaniak/drive/utils/Extensions.kt [hunk 4]
```diff
@@ -103,6 +105,10 @@ typealias IOFile = java.io.File
 typealias IsComplete = Boolean
 typealias Position = Int
 
+fun getAvailableStorageInBytes(): Long = with(StatFs(Environment.getDataDirectory().path)) {
+    return availableBlocksLong * blockSizeLong
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/utils/Extensions.kt:109]:
```suggestion
    return@with availableBlocksLong * blockSizeLong
```

File: app/src/main/res/values-fr/strings.xml [hunk 5]
```diff
@@ -514,6 +514,7 @@
     <string name="noSelectMediaFolders">Aucun dossier sélectionné</string>
     <string name="noSyncFolderNotificationDescription">Cliquez-ici pour configurer les dossiers à synchroniser</string>
     <string name="noSyncFolderNotificationTitle">Aucun dossier sélectionné pour la sauvegarde automatique</string>
+    <string name="notEnoughSpaceAfterDownload">Le téléchargement a été annulé en raison d\'un espace de stockage insuffisant sur votre appareil.</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-fr/strings.xml:517]: Fix this `\`.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/utils/Extensions.kt (hunk 4 above)
Comment (tevincent) [reply app/src/main/java/com/infomaniak/drive/utils/Extensions.kt:109]: Any reason to specify the `@with` here ? We only have 1 line in this function.

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/drive/utils/Extensions.kt:109]: If this code ever moves in the future, the `@with` will help to not make an error and wrongly return to an incorrect scope.

I know it's pretty trivial here, but it's a matter of habit.

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**